packages/regenerator-transform/src/visit.js

```diff
@@ -188,18 +189,34 @@ function getOuterFnExpr(funPath) {
     }
 
     let markDecl = getRuntimeMarkDecl(pp);
-    let markedArray = markDecl.declarations[0].id;
-    let funDeclIdArray = markDecl.declarations[0].init.callee.object;
-    t.assertArrayExpression(funDeclIdArray);
-
-    let index = funDeclIdArray.elements.length;
-    funDeclIdArray.elements.push(node.id);
-
-    return t.memberExpression(
-      markedArray,
-      t.numericLiteral(index),
-      true
-    );
+    let declarations = markDecl.declarations;
+    let index = declarations.length;
+
+    // this can change from pass to pass if something else is altering our body
+    // so we have to look it up every time :/
+    let declarationsIndex = pp.node.body.findIndex((p) => {
```

Comment:
maybe we could gather those declarations in context instead of inserting them right away and insert them all at once in program's exit visitor? 

aint sure if thats better, just an idea